### PR TITLE
#9

### PR DESCRIPTION
**Pull request by alexkrechik**
https://github.com/revjet-qa/wdio-cucumber-steps/issues/9 implementing
Using if pageObjects model
Added one Then step
https://github.com/revjet-qa/wdio-cucumber-steps/issues/10 implementing

### CODE REVIEW

File: cucumber.sh [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env bash
+
+#Run server in background
+http-server ./test/demo-app -p 9000 > /dev/null &
```
Comment (Marketionist) [inline cucumber.sh]: Shouldn't it be `./node_modules/.bin/http-server ./test/demo-app -p 9000 > /dev/null &` ?

File: src/steps/then.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+/* eslint no-undef: 0 */
+
+'use strict'
+
+const { defineSupportCode } = require('cucumber');
+const { pageObject, getPageObject } = require('../helpers/objects.processor')
```
Comment (Marketionist) [inline src/steps/then.js]: `;` is missing at the end of a line

File: src/steps/then.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+const { _r } = require('../helpers/utils')
```
Comment (Marketionist) [inline src/steps/then.js]: `;` is missing at the end of a line

File: src/steps/then.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+module.exports = function () {
+    defineSupportCode(({ Then }) => {
+
+        Then(_r(`^${pageObject} should be present$`), (object) => {
+            /**
+             * The element should be present
+             * @type {PageObject}
+             */
+            const locator = getPageObject(object)
```
Comment (Marketionist) [inline src/steps/then.js]: `;` is missing at the end of a line

File: src/steps/then.js [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+            browser.$(locator).waitForExist();
+        });
+
+    });
+}
```
Comment (Marketionist) [inline src/steps/then.js]: `;` is missing at the end of a line

File: test/features/page_objects/main.js [hunk 6]
```diff
@@ -0,0 +1,6 @@
+'use strict';
+let e = {}
```
Comment (Marketionist) [inline test/features/page_objects/main.js]: `;` is missing at the end of a line

File: test/features/page_objects/main.js [hunk 7]
```diff
@@ -0,0 +1,6 @@
...
+
+e.txtHeader = '//*[@id="header"]';
+
+module.exports = e
```
Comment (Marketionist) [inline test/features/page_objects/main.js]: `;` is missing at the end of a line

File: test/mocha/objects.processor.spec.js [hunk 8]
```diff
@@ -14,18 +14,21 @@ const { _r } = require('../../src/helpers/utils')
...
-            object: `dictionaryObject${dynamicId}`
-        }
+global.pages = {
+    main: {
+        object: `//div[@id='${dynamicId}']`
+    },
+    book: {
+        word: `dictionaryObject${dynamicId}`
     }
 }
```
Comment (Marketionist) [inline test/mocha/objects.processor.spec.js]: `;` is missing at the end of a line

File: test/mocha/objects.processor.spec.js [hunk 9]
```diff
@@ -14,18 +14,21 @@ const { _r } = require('../../src/helpers/utils')
...
 
+global.id = {
+    getId: () => realId
+}
```
Comment (Marketionist) [inline test/mocha/objects.processor.spec.js]: `;` is missing at the end of a line

File: test/mocha/objects.processor.spec.js [hunk 10]
```diff
@@ -14,18 +14,21 @@ const { _r } = require('../../src/helpers/utils')
...
+
+global.objectsProcessor = { }
```
Comment (Marketionist) [inline test/mocha/objects.processor.spec.js]: `;` is missing at the end of a line

File: test/mocha/objects.processor.spec.js [hunk 11]
```diff
@@ -59,7 +62,7 @@ describe('getPageObject', () => {
 
             expect(match).to.be.an('array')
             const strPageObject = match[1]
-            const pageObjectVal = getPageObject.call(World, strPageObject)
+            const pageObjectVal = getPageObject(strPageObject)
```
Comment (Marketionist) [inline test/mocha/objects.processor.spec.js]: `;` is missing at the end of a line

File: test/mocha/objects.processor.spec.js [hunk 12]
```diff
@@ -89,7 +92,7 @@ describe('getDictionaryObject', () => {
 
             expect(match).to.be.an('array')
             const strDictionaryObject = match[1]
-            const dictionaryObjectVal = getDictionaryObject.call(World, strDictionaryObject)
+            const dictionaryObjectVal = getDictionaryObject(strDictionaryObject)
```
Comment (Marketionist) [inline test/mocha/objects.processor.spec.js]: `;` is missing at the end of a line

File: test/wdio.conf.js [hunk 13]
```diff
@@ -188,6 +188,19 @@ exports.config = {
         global.expect = chai.expect;
         global.assert = chai.assert;
         global.should = chai.should();
+
+        global.pages = {
+            main: require('./features/page_objects/main')
+        }
```
Comment (Marketionist) [inline test/wdio.conf.js]: `;` is missing at the end of a line

File: test/wdio.conf.js [hunk 14]
```diff
@@ -188,6 +188,19 @@ exports.config = {
...
+
+        global.id = {
+            value: '',
+            regenerate: () => this.value === (new Date()).getTime(),
+            getId: () => this.value
+        }
```
Comment (Marketionist) [inline test/wdio.conf.js]: `;` is missing at the end of a line

File: test/wdio.conf.js [hunk 15]
```diff
@@ -188,6 +188,19 @@ exports.config = {
...
+
+        global.objectsProcessor = { }
```
Comment (Marketionist) [inline test/wdio.conf.js]: `;` is missing at the end of a line

File: cucumber.sh (hunk 1 above)
Comment (alexkrechik) [reply cucumber.sh]: Agree. Fixed.

File: src/steps/then.js (hunk 2 above)
Comment (alexkrechik) [reply src/steps/then.js]: Lets try to use no semicolons.

File: src/steps/then.js (hunk 4 above)
Comment (alexkrechik) [reply src/steps/then.js]: Lets try to use no semicolons.

File: src/steps/then.js (hunk 5 above)
Comment (alexkrechik) [reply src/steps/then.js]: Lets try to use no semicolons.

File: test/features/page_objects/main.js (hunk 6 above)
Comment (alexkrechik) [reply test/features/page_objects/main.js]: Lets try to use no semicolons.

File: test/features/page_objects/main.js (hunk 7 above)
Comment (alexkrechik) [reply test/features/page_objects/main.js]: Lets try to use no semicolons.

File: test/mocha/objects.processor.spec.js (hunk 8 above)
Comment (alexkrechik) [reply test/mocha/objects.processor.spec.js]: Lets try to use no semicolons.

File: test/mocha/objects.processor.spec.js (hunk 9 above)
Comment (alexkrechik) [reply test/mocha/objects.processor.spec.js]: Lets try to use no semicolons.

File: test/mocha/objects.processor.spec.js (hunk 10 above)
Comment (alexkrechik) [reply test/mocha/objects.processor.spec.js]: Lets try to use no semicolons.

File: test/mocha/objects.processor.spec.js (hunk 12 above)
Comment (alexkrechik) [reply test/mocha/objects.processor.spec.js]: Lets try to use no semicolons.

File: test/wdio.conf.js (hunk 13 above)
Comment (alexkrechik) [reply test/wdio.conf.js]: Lets try to use no semicolons.

File: test/wdio.conf.js (hunk 15 above)
Comment (alexkrechik) [reply test/wdio.conf.js]: Lets try to use no semicolons.

File: src/helpers/objects.processor.js [hunk 16]
```diff
@@ -70,8 +70,8 @@ function getPageObject(str) {
     const pageObjectGetterFunc = objectsProcessor.pageObjectGetter || pageObjectGetter
     const value = pageObjectGetterFunc(str)
     const idValue = value.replace(_r(regDynamicId, 'g'), id.getId())
-    const injection = 'not(ancestor-or-self::*[contains(@style,"visibility: hidden;") ' +
```
Comment (Marketionist) [inline src/helpers/objects.processor.js]: Let's not remove semicolons from CSS `"visibility: hidden;"`

File: test/mocha/objects.processor.spec.js [hunk 17]
```diff
@@ -39,18 +39,18 @@ describe('injectInto', () => {
...
+})
 
 describe('getPageObject', () => {
     const data = [
         {
             step: 'When I click object from main page',
             regExp: _r(`When I click ${pageObject}`),
-            result: `//div[@id='${realId}' and not(ancestor-or-self::*[contains(@style,"visibility: hidden;") ` +
-                'or contains(@style,"display: none;") or contains(@class,"x-hide-offsets")])]'
+            result: `//div[@id='${realId}' and not(ancestor-or-self::*[contains(@style,"visibility: hidden") ` +
```
Comment (Marketionist) [inline test/mocha/objects.processor.spec.js]: Let's not remove semicolons from CSS `"visibility: hidden;"`

File: .gitignore [hunk 18]
```diff
@@ -9,7 +9,6 @@ node_modules
 .*
 !.travis.yml
 !.editorconfig
-!.eslintrc.json
```
Comment (Marketionist) [inline .gitignore]: If we will remove this line, then any changes to `.eslintrc.json` will not be tracked by git - are we sure that it should not be tracked?

**Pull request merged**